cmd/soroban-rpc/lib/preflight/src/fees.rs

```diff
@@ -142,11 +142,11 @@ fn calculate_host_function_soroban_resources(
         })
         .sum();
 
-    // Add a 15% leeway with a minimum of 50k instructions
+    // Add a 20% leeway with a minimum of 50k instructions
     let budget_instructions = budget
         .get_cpu_insns_consumed()
         .context("cannot get instructions consumed")?;
-    let instructions = max(budget_instructions + 50000, budget_instructions * 115 / 100);
+    let instructions = max(budget_instructions + 50000, budget_instructions * 120 / 100);
```

Comment:
Adding this buffer seems like a thing the wallet should do, not preflight. I'd say preflight should return an accurate instruction count as far as it can, then let the wallet figure out how much to pad it.

Reply:
Strictly speaking, the txn should still work with no buffer at all, right? If it doesn't that would mean preflight is counting something wrong.

Reply:
Uhm, fair point @paulbellamy . However, how would the wallet know what to use? (we don't really know either, 20% is a magic number)

(Resolving the thread temporarily so that I can merge)